diam/sm/sm.go

```diff
@@ -96,7 +96,7 @@ func New(settings *Settings) *StateMachine {
 	sm := &StateMachine{
 		cfg:           settings,
 		mux:           diam.NewServeMux(),
-		hsNotifyc:     make(chan diam.Conn),
+		hsNotifyc:     make(chan diam.Conn, 10),
```

Comment:
Were you able to reproduce the problem and test with this change? Also I'm curious for what makes "10" the right size for the buffered channel here.